ViewController.swift

```diff
@@ -7,13 +7,100 @@
     }
 
-
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return people.count
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Person", for: indexPath) as? PersonCell else {
+            fatalError("Enable to deque PersonCell")
+        }
+        let person = people[indexPath.item]
+        cell.name.text = person.name
+        let path = getDocumentsDirectory().appendingPathComponent(person.image)
+        cell.imageView.image = UIImage(contentsOfFile: path.path)
+        cell.imageView.layer.borderColor = UIColor(white: 0, alpha: 0.3).cgColor
+        cell.imageView.layer.borderWidth = 2
+        cell.imageView.layer.cornerRadius = 3
+        cell.layer.cornerRadius = 7
+        
+        return cell
+    }
+    
+    @objc private func addNewPerson() {
+        let picker = UIImagePickerController()
+        picker.allowsEditing = true
+        picker.delegate = self
+        present(picker, animated: true)
+    }
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        guard let image = info[.editedImage] as? UIImage else { return }
+        let imageName = UUID().uuidString
+        let imagepath = getDocumentsDirectory().appendingPathComponent(imageName)
+        
+        if let jpegData = image.jpegData(compressionQuality: 0.8) {
+            try? jpegData.write(to: imagepath)
+        }
+        let person = Person(name: "Unknown", image: imageName)
+        people.append(person)
+        collectionView.reloadData()
+        dismiss(animated: true)
+    }
+    
+    private func getDocumentsDirectory() -> URL {
+        let paths = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
+        return paths[0]
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let person = people[indexPath.item]
+        
+        let alertController = UIAlertController(title: nil, message: "Please Select an Option", preferredStyle: .actionSheet)
+        
+        alertController.addAction(UIAlertAction(title: "Rename", style: .default, handler: { (_) in
+            
+            let renameAlertController = UIAlertController(title: "Rename", message: "What's the new name?", preferredStyle: .alert)
+            renameAlertController.addTextField()
+            
+            renameAlertController.addAction(UIAlertAction(title: "Rename", style: .default) { [weak self, weak renameAlertController] _ in
+                guard let self = self else { return }
+                guard let newName = renameAlertController?.textFields?[0].text else { return }
```

Comment:
This is more of an opinion: it would read better 
```suggestion
                guard let self = self, let newName = renameAlertController?.textFields?[0].text else { return }
```

Less code, less problems :P

Reply:
Oh... That's right Mo already told me that too. 
I forgot to implement. I totally agree that this is a much cleaner and elegant code. 
Fixed! Thank you!